lib/Doctrine/ORM/EntityManager.php

```diff
@@ -353,6 +353,13 @@ public function createQueryBuilder()
      */
     public function flush($entity = null)
     {
+        if (func_num_args() === 1) {
```

Comment:
Consider using > 0

Reply:
Isn't `$entity !== null` simpler? I mean, only the first argument is used anyway

Reply:
Yeah even better

Reply:
That would still allow `flush(null)` though. I'll change it to `!== 0`.